hplot/s2d.go

```diff
@@ -100,16 +103,28 @@ func (pts *S2D) withBand() error {
 			bot[i].X = x
 			bot[i].Y = y - math.Abs(ymin)
 		}
+
 	case HiSteps:
-		for i := range top {
-			// WIP(rmadar): implement the proper band in case of step
-			//              this might actually involve 2xn points - need to think
-			x, y := pts.Data.XY(i)
+		top = make(plotter.XYs, 2*pts.Data.Len())
+		bot = make(plotter.XYs, 2*pts.Data.Len())
+		xerr, ok := pts.Data.(plotter.XErrorer)
+		if !ok {
+			panic("s2d: cannot get X errors, required for HiSteps plotting")
+		}
+
+		// rmadar: the loop here doesn't really need a copy of XYs() but I am not sure
+		// how to handle indices in the loop if only i is defined.
```

Comment:
what do you mean?

Reply:
Let say I don't want to copy data, then I'd have to go with:
```go
for i := range top { // running from 0 to 2n - 1
                       idata: = f(i)  
                       x, y := pts.Data.XY(idata)   
                        xmin, xmax := xerr.XError(idata)
                        ymin, ymax := yerr.YError(idata)
                        top[i] = ...
                        bot[i] = ... 
                }
```
Actually, now I write it, I could probably use something like `f(i) = i // 2`